File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/user/repository/UserOauthLinksRepository.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.user.repository;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.UserOauthLinks;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.Users;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.Optional;
import java.util.UUID;

public interface UserOauthLinksRepository extends JpaRepository<UserOauthLinks, UUID> {

    @Query("SELECT l FROM UserOauthLinks l JOIN FETCH l.user WHERE l.oauthProviderId = :providerId")
    Optional<UserOauthLinks> findByOauthProviderId(String providerId);

    Optional<UserOauthLinks> findByUser(Users users);
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/user/repository/UserOauthLinksRepository.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.user.repository

import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.UserOauthLinks
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.Users
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserOauthLinksRepository : JpaRepository<UserOauthLinks, UUID> {
    fun findByOauthProviderId(providerId: String?): UserOauthLinks?

    fun existsByUser(user: Users): Boolean
}
